corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -320,28 +377,64 @@ hqDefine("geospatial/js/case_grouping_map",[
                 let caseGroup = caseGroupsInstance.getGroupByID(caseGroupID);
                 color = caseGroup.color;
                 const marker = new mapboxgl.Marker({ color: color, draggable: false });  // eslint-disable-line no-undef
-                marker.setLngLat([caseItem.coordinates.lng, caseItem.coordinates.lat]);
+
+                const coordinates = [caseItem.coordinates.lng, caseItem.coordinates.lat];
+                marker.setLngLat(coordinates);
 
                 // Add the marker to the map
                 marker.addTo(map);
                 mapMarkers.push(marker);
+
+                let popupDiv = document.createElement("div");
+                popupDiv.setAttribute("data-bind", "template: 'select-case'");
+
+                let popup = new mapboxgl.Popup({ offset: 25, anchor: "bottom" })  // eslint-disable-line no-undef
+                    .setLngLat(coordinates)
+                    .setDOMContent(popupDiv);
+
+                marker.setPopup(popup);
+
+                const markerDiv = marker.getElement();
+                // Show popup on hover
+                markerDiv.addEventListener('mouseenter', () => marker.togglePopup());
+
+                // Hide popup if mouse leaves marker and popup
+                var addLeaveEvent = function (fromDiv, toDiv) {
+                    fromDiv.addEventListener('mouseleave', function () {
+                        setTimeout(function () {
+                            if (!$(toDiv).is(':hover')) {
+                                // mouse left toDiv as well
+                                marker.togglePopup();
+                            }
+                        }, 100);
+                    });
+                };
+                addLeaveEvent(markerDiv, popupDiv);
+                addLeaveEvent(popupDiv, markerDiv);
+                const colors = {default: color, selected: color};
+
+                const mapMarkerInstance = new mapMarkerModel(caseItem.caseId, caseItem, marker, colors);
+                $(popupDiv).koApplyBindings(mapMarkerInstance);
             }
         });
     }
 
     function uuidv4() {
         // https://stackoverflow.com/questions/105034/how-do-i-create-a-guid-uuid/2117523#2117523
         return "10000000-1000-4000-8000-100000000000".replace(/[018]/g, c =>
-            (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16),
+            (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
         );
     }
 
     function caseGroupSelectModel() {
         'use strict';
         var self = {};
 
-        self.allGroups = ko.observableArray([]);
         self.allCaseGroups;
+        // allGroups and caseGroupsForTable contains the same data, but there's weird knockoutjs behaviour
+        // if we're making allGroups an observable. caseGroupsForTable is populated by setCaseGroupsForTable
```

Comment:
I'm curious about this, what is the weird behaviour if `allGroups` is an `observableArray`?

Reply:
When `allGroups` is an observable an issue happened where groups would get updated in the background seemingly randomly when you show only a subset of groups on the screen and then select to show all groups on the screen. The groups that was obscured would then either have all been unassigned or mostly unassigned from their initial groups. 

I haven't been able to figure out exactly why, as the logic for handling the selected groups on the screen seems solid, so my conclusion was that somewhere something listens to this observable which triggers a chain reaction which is hard to follow.

Reply:
Hmm... yeah this definitely smells of some funky JS behaviour. :thinking: 

Since JS handles an array as pass-by-reference, I wonder if the array was perhaps being pased somewhere else as well and, instead of copying the array, it was modifying the old reference directly.